### DISCUSSION TITLE
Ability to add issues from repositories belonging to other Organization/Owner

### DISCUSSION THREAD

**Discussion by timharris777** [community/community#6212] [Projects and Issues] [answered]
My ream is trying to use github projects (issues beta) to track all of our work. We currently have issues in multiple repositories across multiple organizations. We are currently are only able to add issues to the project that are in the same organization as the project. This prevents us from being able to track "all" of our work in a single project.

**Comment by Brend-Smits** [comment]
This is a much-needed feature. 
We want to track our projects in an internal (InnerSource) project board with issues from both internal repositories and open source repositories. In the current project board beta, this is still impossible. 
There's more demand for this feature, see: https://github.com/dear-github/dear-github/issues/209

**Comment by CatalinaMoisuc** [comment]
I subscribe to this request!! I really like what you are doing now with these boards, keep up the good work!! but this is critical for us.
We tried adding the link to the issue via a note in the old boards, but really, way too much manual work.

**Comment by neuromusic** [comment]
+1 this would be incredibly useful for the work we do and get us much closer to dropping Zenhub

**Comment by tmigone** [comment]
+1 to this. It's the only thing holding me back from using GitHub projects.

**Comment by lahabana** [comment]
+1 to this. We contribute a lot to an OSS project in an external org and it would enable us to track everything in one place!

**Comment by rkoster** [comment]
This seems to work from the web interface, `CTRL + Space` followed by the full URL to an issue seems to work for me. When using the web inspector I can see a API call being made which has an URL field for the project item. 

Unfortunately I was not yet able to achieve the same using the graphsql API. It is lacking the URL field here: https://docs.github.com/en/graphql/reference/input-objects#addprojectnextiteminput.

**Reply by tmigone** [reply]
Looks like it was recently added, awesome!

**Reply by galargh** [reply]
If you'd like to be able to use full GitHub search syntax to find issues cross-org and add them in bulk, I created an extension just for that - https://chrome.google.com/webstore/detail/gadder/lplkcdhlldoofpdgngieldiihjcdcmdi

To use this, you have to install the extension and type `q=` followed by a GitHub issue seach query in the input box of a GitHub Project (Beta).

**Reply by galargh** [reply]
Here's the code for the extension if you're interested - https://github.com/galargh/gadder

**Reply by tim-rohrer** [reply]
@tmigone, So I’m still not understanding the solution. This isn’t automatic, right? What you’re describing is a manual way to add a link to an issue in an organization in which we are a member, correct?

What I would like to see is a work flow such that if I’m listed as a reviewer for a PR in one of my organizations, I can have that automatically pop up the project board. 

@galargh’s extension might be of use, but I’m not using Chrome. 

Cheers!

**Reply by augustjk** [reply]
Pertaining to the top level post about adding issues from other orgs via the graphql API, the mutation [addProjectV2DraftIssue](https://docs.github.com/en/graphql/reference/mutations#addprojectv2draftissue) with the html url of the issue in the `title` field does the trick. Thanks to @aomarks for testing and confirming this!

**Reply by giohappy** [reply]
thanks for sharing that workaround @augustjk. Of course this is only a partial solution, since it doesn't build a complete relationship netween the issue and the project as mentioned in https://github.com/orgs/community/discussions/6212#discussioncomment-3862195

**Comment by Jille** [comment]
As a workaround, you could use [SnoozeThis](https://github.com/apps/snoozethis) to create tracking issues in your own project that automatically get closed once the issue you're waiting for is done.

Simply create a new issue with the description:
```
@SnoozeThis <url of the issue in another project> -> closed
```

**Comment by AgeManning** [comment]
It would also be useful to track issues of forked repositories. 

I.e we have forked a repository and creating issues on our fork.

**Comment by tayloramurphy** [comment]
My experience on this is that you can add an issue from one org to another orgs project, but there's no indication in the Issue UI that the issue is a part of another project. That makes it really frustrating as it's hard to tell what is an isn't already in one's project.

**Comment by rileybroughten** [comment, marked as answer]
You are now able to add items from different organizations to your project - check out the announcement [here](https://github.blog/changelog/2023-02-23-github-issues-projects-february-23rd-update/#add-cross-organization-issues-and-pull-requests-to-projects)!

**Reply by ssbarnea** [reply]
@rileybroughten Sorry but while an outside issue can be added to a project from the project board, that does not mark it as assigned to any project from its own issue page, making impossible to know which where planned or not.  Basically the only place to see that is on project board and that makes the entire feature unsuitable for practical use.

![](https://sbarnea.com/ss/Screen-Shot-2023-05-19-06-59-31.58.png)

**Reply by farzadpanahi** [reply]
"Adding" an issue from a different org works now. That is great. But it looks like there is a bug when you want to "create" an issue for a repo in a different org from the Project page.

Here you can see I have the option to create a new issue on `edgepi-rpc-server` repo:
![image](https://github.com/community/community/assets/3793563/aaf1b20b-d71d-4781-a20c-e992b2023f96)

But when I click on "Create new issue" it would open a modal window which has the wrong repo `edgepi-python-sdk` selected, with no option to choose the original repo `edgepi-rpc-server`:
![image](https://github.com/community/community/assets/3793563/bca6e51f-0bb0-42c6-ae73-9af9273b9ed4)

Note that repos `edgepi-rpc-server` and  `edgepi-python-sdk`  belong to two different orgs and the project is created on the same org where  `edgepi-python-sdk` repo belongs to. 

So my workaround at the moment is to go and create issues under `edgepi-rpc-server`  directly and then come back to the Project page on the different repo and add those already created issues. That works but it is a bit inconvenient. 

@rileybroughten Is this a bug or expected behavior?

**Reply by rileybroughten** [reply]
@farzadpanahi are you still experiencing this behavior where it opens the dialog with a different repository pre-filled than what you had typed out (a repository from another organization)?

**Reply by farzadpanahi** [reply]
@rileybroughten I just checked this. I think it is fixed. Thanks : )

**Comment by florent-solt-wmx** [comment]
@rileybroughten via the graphql api (a query like `search(label:X project:org/number)`), I can see issues from a different org, all good.
But! I can't see custom fields via `projectItems`? It only works for issues in the same org as the project.
How should I do? I mean, to see the same information in my project view and via graphql.

**Comment by marty-downs** [comment]
This is moving in a good direction, but still pretty clunky. What would be most useful here is to be able to assign existing tasks in one project to a project in a different repository (not in the same organization) via the issue sidebar. It's a great feature within a single organization, but challenging for working in multiple contexts at once.

**Comment by maliberty** [comment]
Is it possible to use cross-org repos with the "Auto-add to project" workflow?  I haven't found a way.  Thanks.

**Comment by lwld** [comment]
Cool, just found this :sunglasses: I was looking for it a while ago already but didn't figure how it works and that it exists.

It's not very discoverable, you have to "know" about it - it could also mention "organization" in the + Add item field.

Also improvement would be to be able to go the other way - from an issue/PR in one organization, in the sidebar under Project, it would be cool to be able to search for outside organization Projects.

Even better would be to be able to link the Project in a different organization's repo, just as it can be done across repos within the orga.

**Reply by reteps** [reply]
> Also improvement would be to be able to go the other way - from an issue/PR in one organization, in the sidebar under Project, it would be cool to be able to search for outside organization Projects.

> Even better would be to be able to link the Project in a different organization's repo, just as it can be done across repos within the org.

Both of these would be awesome!
